pkg/lib/cockpit-components-table.jsx

```diff
@@ -75,9 +75,36 @@ export const ListingTable = ({
     let rows = tableRows;
 
     const [expanded, setExpanded] = useState({});
+    const [newItems, setNewItems] = useState([]);
+    const [currentRowsKeys, setCurrentRowsKeys] = useState([]);
     const [activeSortIndex, setActiveSortIndex] = useState(sortBy ? sortBy.index : 0);
     const [activeSortDirection, setActiveSortDirection] = useState(sortBy ? sortBy.direction : SortByDirection.asc);
 
+    useEffect(() => {
+        const getRowKeys = rows => {
+            const keys = [];
+
+            rows.forEach(row => {
+                if (row.props && row.props.key)
+                    keys.push(row.props.key);
+            });
+
+            return keys;
+        };
+
+        const new_keys = getRowKeys(rows);
+        if (JSON.stringify(new_keys) === JSON.stringify(currentRowsKeys))
+            return;
+        // Don't highlight all when the list gets loaded
+        if (currentRowsKeys.length !== 0) {
+            new_keys.forEach(key => {
+                if (currentRowsKeys.indexOf(key) === -1)
+                    setNewItems([...newItems, key]);
```

Comment:
Is it wise to call a state set function that often? Shouldn't it rather do this once after the loop?

(And then use filtering on new_keys instead of forEach)